zenoh/src/net/routing/hat/router/network.rs

```diff
@@ -756,7 +757,7 @@ impl Network {
                 tracing::trace!("Update edge (link) {} {}", self.graph[self.idx].zid, zid);
                 self.update_edge(self.idx, idx);
             }
-            self.graph[self.idx].links.push(zid);
+            self.graph[self.idx].links.insert(zid);
```

Comment:
Same comment as above on adding a debug log message is `insert(zid)` returns `Some`?